File: TypeScript/codewars/Typescript/Deodorant-Evaporator.ts
export function evaporator(content: number, evapPerDay: number, threshold: number): number {
	let quantityOfDays = 0;
	let vapor = content;
	const percentThreshold = (content * threshold) / 100;

	while (vapor > percentThreshold) {
		vapor = vapor - (vapor * evapPerDay) / 100;

		quantityOfDays++;
	}

	return quantityOfDays;
}

File: Csharp/CodeWars/src/Completed/DeodorantEvaporator.cs
namespace CodeWars.Completed;

public class DeodorantEvaporator
{
    public static int Evaporator(double content, double evap_per_day, double threshold)
    {
        int quantityOfDays = 0;
        double percentThreshold = content * threshold / 100;

        while (content > percentThreshold)
        {
            content -= content * evap_per_day / 100;

            quantityOfDays++;
        }

        return quantityOfDays;
    }
}
